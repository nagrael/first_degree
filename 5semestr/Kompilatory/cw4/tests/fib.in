int max = 15;
int i = 0;


int fib_iter(int n) {

int i = 0;

    if(n <= 0) {
        return 0;
    }
    if(n == 1) {
        return 1;
    }
    else
    {
        int a = 0, b = 1, sum = 0, i = 0;
        while(i < (n-1)) {
            sum = a + b;
            a = b;
            b = sum;
            i = i+1;
        }
        return sum;
    }
}


int fib(int n) {
    if(n <= 0) {
        return 0;
    }
    if(n == 1) {
        return 1;
    } 
    else {
        return fib(n-1) + fib(n-2);
    }
}


int trib(int n) {
    if(n <= 1) {
        return 0;
    }
    if(n == 2) {
        return 1;
    } 
    else {
        return trib(n-1) + trib(n-2) + trib(n-3);
    }
}


i = 0;
while (i <= max ) {
    print fib(i);
    i = i + 1;
}

i = 0;
while(i <= max) {
    print fib_iter(i);
    i = i+1;
}

i = 0;
while(i <= max) {
    print trib(i);
    i = i+1;
}

