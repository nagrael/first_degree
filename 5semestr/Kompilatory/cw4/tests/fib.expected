0
1
1
2
3
5
8
13
21
34
55
89
144
233
377
610
0
1
1
2
3
5
8
13
21
34
55
89
144
233
377
610
0
0
1
1
2
4
7
13
24
44
81
149
274
504
927
1705
